apis/privilegedaccessmanager/v1beta1/privilegedaccessmanagerentitlement_types.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var PrivilegedAccessManagerEntitlementGVK = GroupVersion.WithKind("PrivilegedAccessManagerEntitlement")
+
+// PrivilegedAccessManagerEntitlementSpec defines the desired state of
+// PrivilegedAccessManagerEntitlement.
+// +kcc:proto=google.privilegedaccessmanager.v1.Entitlement
+// +required
+type PrivilegedAccessManagerEntitlementSpec struct {
+	// Immutable. The Project that this resource belongs to.
+	// One and only one of 'projectRef', 'folderRef', or 'organizationRef' must
+	// be set.
+	// +optional
+	ProjectRef refs.ProjectRef `json:"projectRef,omitempty"`
+
+	// Immutable. The Folder that this resource belongs to.
+	// One and only one of 'projectRef', 'folderRef', or 'organizationRef' must
+	// be set.
+	// +optional
+	FolderRef refs.FolderRef `json:"folderRef,omitempty"`
+
+	// Immutable. The Organization that this resource belongs to.
+	// One and only one of 'projectRef', 'folderRef', or 'organizationRef' must
+	// be set.
+	// +optional
+	OrganizationRef refs.OrganizationRef `json:"organizationRef,omitempty"`
+
+	// Immutable. Location of the resource.
+	// +required
+	Location *string `json:"location"`
+
+	// The PrivilegedAccessManagerEntitlement name. If not given, the
+	// 'metadata.name' will be used.
+	// +optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Optional. Who can create grants using this entitlement. This list should
+	// contain at most one entry.
```

Comment:
Do you mean this accepts a list with no more than one value?

Reply:
Looks like so: https://cloud.google.com/iam/docs/reference/pam/rest/v1beta/organizations.locations.entitlements#resource:-entitlement